kernel/src/main/java/org/kframework/ksearchpattern/KSearchPatternFrontEnd.java

```diff
@@ -0,0 +1,122 @@
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Frontend for k-compiled-search-pattern tool.
+ * <p>
+ * k-compile-search-patterrn is used by the new krun frontend in order to
+ * convert a search pattern written as a rule bubble into a KORE search pattern
+ * for the Haskell backend.
+ */
+public class KSearchPatternFrontEnd extends FrontEnd {
+
+    private final KSearchPatternOptions options;
+    private final Provider<KompileOptions> kompileOptions;
+    private final KExceptionManager kem;
+    private final Provider<FileUtil> files;
+    private final GlobalOptions globalOptions;
+    private final DefinitionScope scope;
+    private final Provider<File> kompiledDir;
+    private final Provider<CompiledDefinition> compiledDef;
+
+    @Inject
+    public KSearchPatternFrontEnd(
+            KSearchPatternOptions options,
+            KExceptionManager kem,
+            Provider<KompileOptions> kompileOptions,
+            GlobalOptions globalOptions,
+            @Usage String usage,
+            @ExperimentalUsage String experimentalUsage,
+            JarInfo jarInfo,
+            Provider<FileUtil> files,
+            @KompiledDir Provider<File> kompiledDir,
+            Provider<CompiledDefinition> compiledDef,
+            DefinitionScope scope) {
+        super(kem, globalOptions, usage, experimentalUsage, jarInfo, files);
+        this.options = options;
+        this.kompileOptions = kompileOptions;
+        this.globalOptions = globalOptions;
+        this.kem = kem;
+        this.files = files;
+        this.scope = scope;
+        this.kompiledDir = kompiledDir;
+        this.compiledDef = compiledDef;
+    }
+
+    public static List<Module> getModules() {
+        List<Module> modules = new ArrayList<>();
+        modules.add(new KSearchPatternModule());
+        modules.add(new JCommanderModule());
+        modules.add(new CommonModule());
+        return modules;
+    }
+
+    @Override
+    protected int run() {
+        scope.enter(kompiledDir.get());
+        try {
+          FileUtil files = this.files.get();
+          CompiledDefinition compiledDef = this.compiledDef.get();
+          KompileOptions kompileOptions = this.kompileOptions.get();
+          Rule pattern = compiledDef.compilePatternIfAbsent(files, kem, options.pattern(), Source.apply("<command line>"));
+          K patternTerm = RewriteToTop.toLeft(pattern.body());
+          K patternCondition = pattern.requires();
+          org.kframework.definition.Module mod = compiledDef.executionModule();
+          ModuleToKORE converter = new ModuleToKORE(mod, files, compiledDef.topCellInitializer, kompileOptions);
+          StringBuilder sb = new StringBuilder();
+          ExpandMacros macroExpander = ExpandMacros.forNonSentences(mod, files, kompileOptions, false);
+          K withMacros = macroExpander.expand(patternTerm);
+          K kWithInjections = new AddSortInjections(mod).addInjections(withMacros);
```

Comment:
This looks like a very short `kompile` pipeline? Could it be framed as such? Mostly curious.

Reply:
What do you mean by "framed as"? Like, what change are you suggesting? It is correct that we are essentially taking a single pattern with a single optional side condition and parsing and kompiling it to kore.

Reply:
Not suggesting a change, just want to understand (should have made that clear). The main reason I'm thinking is because we've discussed changes to the kompile pipeline (ie framing it more declaratively using dependencies/checks interface), and if that happens the question is should this also happen here, or not?

Nothing actionable, just want to make sure it's been noted.